### DISCUSSION TITLE
spread syntax in .find() method in mongoose 7.0.2

### DISCUSSION THREAD

**Discussion by fedechaves** [Automattic/mongoose#13194] [Q&A] [answered]
Hi everyone! When working with mongoose 7.0.2, this code stops working and no matter the method, I can't use spread syntax as a query, it always returns an empty array.

the code **(works with "mongoose": "^6.3.1")**: 
```
export const getHotels = async (req, res, next) => {
  const { min, max, ...others } = req.query;
  try {
    const hotels = await Hotel.find({
      ...others,
      cheapestPrice: { $gt: min | 1, $lt: max || 999 },
    }).limit(req.query.limit);
    res.status(200).json(hotels);
  } catch (err) {
    next(err);
  }
};
```
the query:  localhost:8800/api/hotels?featured=true&limit=4
what console.log(others) returns: 
{ featured: 'true', limit: '4' }

does anyone know how to fix it or what I'm doing wrong?

**Comment by vkarpov15** [comment, marked as answer]
`const { min, max, limit, ...others } = req.query;` . I suspect that `limit` is ending up in your query filter. This is due to the [strictQuery changes in Mongoose 7](https://mongoosejs.com/docs/migrating_to_7.html#strictquery).

**Comment by fedechaves** [comment]
thank you @vkarpov15 !
